### PR TITLE
[cloud-provider-yandex] Add diskType parameter for masterNodeGroup and nodeGroups

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

Added `diskType` parameter for `masterNodeGroup` and `nodeGroups`.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-yandex
type: feature
summary: Added diskType parameter for masterNodeGroup and nodeGroups in YandexClusterConfiguration.
```

### CODE REVIEW

**Comment by meatuses** [comment]
https://github.com/deckhouse/deckhouse/issues/8411

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 9187678346 9187678346-1-con-1-25 /sys/deckhouse-oss/install:pr8384 redos@158.160.99.86

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.25 9187678346 9187678346-1-con-1-25 /sys/deckhouse-oss/install:pr8384 redos@158.160.99.86

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: candi/cloud-providers/yandex/openapi/cluster_configuration.yaml [hunk 1]
```diff
@@ -124,8 +124,15 @@ apiVersions:
...
-                example: 20
+                example: network-ssd
                 x-doc-default: 50
+              diskType: &instanceClassDiskType
+                type: string
+                description: |
+                  Yandex Compute Instance disk type.
+                example: network-ssd-io-m3
+                x-doc-default: network-ssd
+                enum: [ network-ssd, network-ssd-io-m3 ]
```
Comment (RomanenkoDenys) [inline candi/cloud-providers/yandex/openapi/cluster_configuration.yaml:135]: may be we need to add all disk types there ?

**Review by RomanenkoDenys — COMMENTED** [review]
one question

**Review by name212 — COMMENTED** [review]
question from Denis

**Review by z9r5 — APPROVED** [review]

File: candi/cloud-providers/yandex/openapi/cluster_configuration.yaml (hunk 1 above)
Comment (vitaliy-sn) [reply candi/cloud-providers/yandex/openapi/cluster_configuration.yaml]: We should not allow HDD to be used. The remaining ssd network-ssd-nonreplicated is added to the enum.

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**